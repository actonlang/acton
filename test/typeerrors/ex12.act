actor main(env):
    env.exit()
    
