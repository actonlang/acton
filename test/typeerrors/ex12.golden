Building file ../test/typeerrors/ex12.act
  Compiling ex12.act for release

ERROR: Error when compiling ex12 module: Type error

Cannot satisfy the following simultaneous constraints for the unknown types t0, t1, t2:

  |
2 |    env.exit()
  |    ^^^
The type of env ( _) must have an attribute exit with type t0() -> _; no such type is known.

  |
2 |    env.exit()
  |    ^^^^^^^^^^
t0 must be a subclass of t1

  |
2 |    env.exit()
  |    ^^^^^^^^^^
t1 must implement $Wrapped[_, t2]

  |
2 |    env.exit()
  |    ^^^^^^^^^^
The type of env.exit() (which we call t2) must be a subclass of proc

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

